### ISSUE TITLE
Problem resolving $ref in some cases

### ISSUE THREAD

**Issue by sw23** [coveooss/json-schema-for-humans#57]
I'm seeing an issue resolving references in the case where a schema node is created normally (i.e. not from a reference) and then a single property references that node.  In `_resolve_ref`, the code that looks for existing references falls through to the `else` clause below even when 'other_user' is None (the reference wasn't created by another reference user but rather an explicit use).
```python
else:
    # Both nodes are the same depth. The other having been seen first,
    # this node will be hidden and linked to the other node
    current_node.is_displayed = False
    return other_user
```
So this line:
https://github.com/coveooss/json-schema-for-humans/blob/4909dbe1db6efb3acd2b7c430f8c1650b48c5078/json_schema_for_humans/generate.py#L356

Returns None however it should return `found_reference` in this case (I think).  Replacing `return other_user` with `return found_reference` fixed it for me but seems like it would break what the `else` clause was intended to handle.

To recreate:
```
generate-schema-doc --expand-buttons ./tests/cases/references.json
```
- Go to references.html#anchor_nested_reference
- You will see `type: object` instead of `type: string` (because the reference points to a string definition)

I'm not super confident in my fix but can attempt a PR if that'd be helpful.  Thanks!

**Comment by dblanchette** [comment]
Thanks for taking the time to investigate this and provide steps to reproduce, it's really appreciated!

In this case, there is no `other_user`, so I added a condition that `other_user` exists and is a node that references another (which should be the case since it is in `reference_users_for_this_schema`, but we can never be too cautious). If `other_user` exists, we still want to hide the current node as to make sure elements are only documented once.

This should now be fixed in version 0.19.4 that just released.

**Comment by sw23** [comment]
Thank you!!
